### PR TITLE
Slide to open maps

### PR DESCRIPTION
**Pull request by kevintbradbury**
BlackSourceLabs/BlackNectar-iOS#54 Slide to Open Maps
Adds an extension to StoresTableView to utilize slide transitions
between TableView and MapView

### CODE REVIEW

File: BlackNectar/BlackNectar/StoresTableViewController.swift [hunk 1]
```diff
@@ -59,6 +61,8 @@ class StoresTableViewController: UITableViewController, SideMenuFilterDelegate {
             
         }
         
+        setsEdgePanGesture()
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/StoresTableViewController.swift]: Function name should be an imperative-sounding verb. AKA, `setEdgeGesture`

File: BlackNectar/BlackNectar/StoresTableViewController.swift [hunk 2]
```diff
@@ -256,8 +261,83 @@ extension StoresTableViewController {
...
+    
+}
 
+extension StoresTableViewController {
+    
+    func setsEdgePanGesture() {
+        
+        edgeGesture.addTarget(self, action: #selector(self.handleRightEdge(gesture:)))
+        edgeGesture.edges = .right
+        edgeGesture.delegate = self
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/StoresTableViewController.swift:270]: Very good!!

File: BlackNectar/BlackNectar/StoresTableViewController.swift [hunk 3]
```diff
@@ -256,8 +261,83 @@ extension StoresTableViewController {
...
+        
+        self.view.addGestureRecognizer(edgeGesture)
+        
+    }
+    
+    func handleRightEdge(gesture: UIScreenEdgePanGestureRecognizer) {
+        
+        switch gesture.state {
+            
+        case .began, .changed:
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/StoresTableViewController.swift:280]: `case` needs to be indented from `switch`, even though XCode does not do it for you.
That is a standard practice accross the industry.

File: BlackNectar/BlackNectar/StoresTableViewController.swift [hunk 4]
```diff
@@ -256,8 +261,83 @@ extension StoresTableViewController {
             }
             
         }
-    
+        
     }
+    
+}
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/StoresTableViewController.swift:263]: Add 
```//MARK - Gesture Code ```

File: BlackNectar/BlackNectar/StoresTableViewController.swift [hunk 5]
```diff
@@ -256,8 +261,83 @@ extension StoresTableViewController {
...
+        
+    }
+    
+    func handleRightEdge(gesture: UIScreenEdgePanGestureRecognizer) {
+        
+        switch gesture.state {
+            
+        case .began, .changed:
+            
+            if !panningWasTriggered {
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/StoresTableViewController.swift]: Recommendation: Put this code in a different function. Right now there is a lot of code happening in a switch statement.

File: BlackNectar/BlackNectar.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -24,7 +24,6 @@
 		327BAD661DFA26AE000EA25E /* RoundedButtonView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 327BAD651DFA26AE000EA25E /* RoundedButtonView.swift */; };
 		32B079791DEF8B7A00FACD1B /* StoresTableViewCell.swift in Sources */ = {isa = PBXBuildFile; fileRef = 32B079781DEF8B7A00FACD1B /* StoresTableViewCell.swift */; };
 		32E134411E15FE6700FD65DC /* AromaSwiftClient.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 32E1343C1E15FE6700FD65DC /* AromaSwiftClient.framework */; };
-		32E134421E15FE6700FD65DC /* AromaSwiftClient.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 32E1343C1E15FE6700FD65DC /* AromaSwiftClient.framework */; };
```
Comment (SirWellington) [inline BlackNectar/BlackNectar.xcodeproj/project.pbxproj:29]: Why does it look like Aroma is being removed?

**Review by SirWellington — COMMENTED** [review]
Solid Snake :snake:

File: BlackNectar/BlackNectar/StoresTableViewController.swift (hunk 3 above)
Comment (kevintbradbury) [reply BlackNectar/BlackNectar/StoresTableViewController.swift:280]: I noticed that, I can amend that post merge in Xcode.

File: BlackNectar/BlackNectar/StoresTableViewController.swift (hunk 1 above)
Comment (kevintbradbury) [reply BlackNectar/BlackNectar/StoresTableViewController.swift]: 👍🏻

File: BlackNectar/BlackNectar.xcodeproj/project.pbxproj (hunk 6 above)
Comment (kevintbradbury) [reply BlackNectar/BlackNectar.xcodeproj/project.pbxproj:29]: It still shows up in the project file, possibly from when I had to reload the frameworks from Carthage?

Comment (SirWellington) [reply BlackNectar/BlackNectar.xcodeproj/project.pbxproj:29]: Got it. I see.

**Pull request merged**